routes/freelance/router.js

```diff
@@ -0,0 +1,99 @@
+/** @module freelance/router */
+'use strict';
+
+var express = require('express');
+var router = express.Router();
+var middleware =  require('../middleware');
+const utils = require('../utils');
+var mongoose = require('mongoose');
+var ObjectId = mongoose.Types.ObjectId;
+const Freelance = mongoose.model('Freelance');
+const Review = mongoose.model('Review');
+
+// Supported methods.
+router.all('/', middleware.supportedMethods('GET, POST, PUT, OPTIONS'));
+router.all('/new', middleware.supportedMethods('GET, OPTIONS')); //add delete later
+router.all('/:freelanceid', middleware.supportedMethods('GET, PUT, OPTIONS')); //add delete later
+
+// GET /freelance/new
+router.get('/new', function(req, res, next) {
+  if (req.accepts('text/html')) {
+    res.render('freelancer-create', {
+      title: "JobAdvisor - Create Freelancer Profile" ,
+    });
+  } else res.sendStatus(400);
+});
+
+// GET freelance/:freelanceid
+router.get('/:freelanceid', function(req, res, next) {
+
+  if (ObjectId.isValid(req.params.freelanceid)) {
+    // distinguish between raw and ajax GET request (to render page or return JSON)
+    if(req.headers.ajax) {
```

Comment:
As said, we should consider splitting this into two routes `/freelance/:id` and `/freelance/:id/view`